From: Stefan Haberland <sth@linux.vnet.ibm.com>
Subject: s390/dasd: make query host access interruptible
Patch-mainline: v4.9-rc1
Git-commit: f50af850f407df7b548802461bc248c7683c6dd3
References: bnc#1038912, LTC#153126

Description:  dasd: make query host access interruptible
Symptom:      If the DASD device gets blocked for any reason,
              e.g. because it is reserved somewhere, the 
              host_access_count sysfs entry or the host_access_list
              debugfs entry may sleep forever. 
Problem:      The respective function does not use the interruptible 
              sleep_on function.
Solution:     Use the interruptible sleep_on function so that userspace
              can use ^C to abort the operation.
Reproduction: Use the host_access_count sysfs entry or the
              host_access_list debugfs entry on a reserved DASD.

Upstream-Description:

              s390/dasd: make query host access interruptible

              If the DASD device gets blocked for any reason, e.g. because it is reserved
              somewhere, the host_access_count sysfs entry or the host_access_list
              debugfs entry may sleep forever. Make it interruptible so that userspace
              can use ^C to abort the operation.

              Signed-off-by: Stefan Haberland <sth@linux.vnet.ibm.com>
              Signed-off-by: Martin Schwidefsky <schwidefsky@de.ibm.com>


Signed-off-by: Stefan Haberland <sth@linux.vnet.ibm.com>
Acked-by: John Jolly <jjolly@suse.com>
---
 drivers/s390/block/dasd_eckd.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/s390/block/dasd_eckd.c
+++ b/drivers/s390/block/dasd_eckd.c
@@ -5244,7 +5244,7 @@ static int dasd_eckd_query_host_access(s
 
 	cqr->buildclk = get_tod_clock();
 	cqr->status = DASD_CQR_FILLED;
-	rc = dasd_sleep_on(cqr);
+	rc = dasd_sleep_on_interruptible(cqr);
 	if (rc == 0) {
 		*data = *host_access;
 	} else {
